### PR TITLE
[db] Define PersonalAccessToken model for TypeORM

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Adds DB model into server to read PATs.

This is needed to be able to authenticate, and authorize, requests containing the PAT

## Related Issue(s)

* Part of https://github.com/gitpod-io/gitpod/issues/14912

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by geropl** [comment]
@easyCZ I bet you already weighted the options, but: Why not have a GRPC service that is consumed by server, instead of giving it DB-level access?

**Comment by easyCZ** [comment]
TL;DR: I consider server being a reader of the shared table a lesser evil compared to having an "internal but unauthenticated" endpoint which has access to these tokens.

Primary reason is we wouldn't be able to properly authorize this endpoint. One could argue that we could keep the endpoint private but I find this to be greater risk than having server be a reader (it wouldn't write) of the shared DB table.

This largely stems from "private unathenticated endpoints" are a reasonable security risk as they can get exposed by accident without necessarily intending. 

I anticipate that with the work on standing up an Identity component (SSO + Fine Grained Auth), it would then move there. and be removed from server.

**Comment by easyCZ** [comment]
Spoke sync with @geropl, and he's okay with this approach.

**Review by geropl — APPROVED** [review]
Change LGTM.
